docs/deploy/Deploy-DependentResources.bicep

```diff
@@ -0,0 +1,124 @@
+/*
+This template deploys the following:
+* Storage account (if enableTemplateStoreIntegration is set to true)
+* Storage account container (if enableTemplateStoreIntegration is set to true)
+* Key vault (if deployKeyVault is set to true)
+* User assigned identity with Key Vault Secrets User role on the Key Vault (if deployKeyVault is set to true)
+* Role assignment for the user assigned identity to access the Key Vault (if deployKeyVault is set to true)
+*/
+
+@description('Location where the storage account is deployed. For list of Azure regions where Blob Storage is available, see [Products available by region](https://azure.microsoft.com/en-us/explore/global-infrastructure/products-by-region/?products=storage).')
+@allowed([
+  'australiacentral'
+  'australiaeast'
+  'australiasoutheast'
+  'brazilsouth'
+  'canadacentral'
+  'canadaeast'
+  'centralindia'
+  'centralus'
+  'chinaeast2'
+  'chinanorth2'
+  'chinanorth3'
+  'eastasia'
+  'eastus'
+  'eastus2'
+  'francecentral'
+  'germanywestcentral'
+  'italynorth'
+  'japaneast'
+  'japanwest'
+  'koreacentral'
+  'northcentralus'
+  'northeurope'
+  'norwayeast'
+  'polandcentral'
+  'qatarcentral'
+  'southafricanorth'
+  'southcentralus'
+  'southeastasia'
+  'southindia'
+  'swedencentral'
+  'switzerlandnorth'
+  'uaenorth'
+  'uksouth'
+  'ukwest'
+  'westcentralus'
+  'westeurope'
+  'westus'
+  'westus2'
+  'westus3'
+])
+param location string
+
+@description('If set to true, a storage account and blob container will be deployed with the specified names for storing custom templates.')
+param deployTemplateStore bool
+
+@description('Name of the storage account to be deployed.')
+param templateStorageAccountName string

```

Comment:
If `deployTemplateStore` is set to false, doesn't make sense to require the names so we should have empty defaults.